homeassistant/components/binary_sensor/w800rf32.py

```diff
@@ -0,0 +1,161 @@
+"""
+Support for w800rf32 binary sensors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.w800rf32/
+
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASSES_SCHEMA, PLATFORM_SCHEMA, BinarySensorDevice)
+from homeassistant.components.w800rf32 import (
+    W800RF32_DEVICE, CONF_FIRE_EVENT, CONF_OFF_DELAY)
+from homeassistant.const import (
+    ATTR_ENTITY_ID, ATTR_STATE,
+    CONF_DEVICE_CLASS, CONF_NAME,
+    ATTR_NAME, CONF_DEVICES)
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers import event as evt
+from homeassistant.util import dt as dt_util
+from homeassistant.helpers.dispatcher import (async_dispatcher_connect)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['w800rf32']
+
+ATTR_FIRE_EVENT = 'fire_event'
+EVENT_BUTTON_PRESSED = 'button_pressed'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_DEVICES, default={}): {
+        cv.string: vol.Schema({
+            vol.Optional(CONF_NAME): cv.string,
+            vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
+            vol.Optional(CONF_FIRE_EVENT, default=False): cv.boolean,
+            vol.Optional(CONF_OFF_DELAY):
+            vol.Any(cv.time_period, cv.positive_timedelta)
+        })
+    },
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup_platform(hass, config,
+                               add_entities, discovery_info=None):
+    """Set up the Binary Sensor platform to w800rf32."""
+    binary_sensors = []
+    # device_id --> "c1 or a3" X10 device. entity (type dictionary)
+    # --> name, device_class etc
+    for device_id, entity in config[CONF_DEVICES].items():
+
+        _LOGGER.debug("Add %s w800rf32.binary_sensor (class %s)",
+                      entity[ATTR_NAME], entity.get(CONF_DEVICE_CLASS))
+
+        device = W800rf32BinarySensor(
+            hass, device_id, entity.get(CONF_NAME),
+            entity.get(CONF_DEVICE_CLASS),
+            entity[CONF_FIRE_EVENT], entity.get(CONF_OFF_DELAY))
+
+        binary_sensors.append(device)
+
+    add_entities(binary_sensors)
+
+
+class W800rf32BinarySensor(BinarySensorDevice):
+    """A representation of a w800rf32 binary sensor."""
+
+    def __init__(self, hass, device_id, name, device_class=None,
```

Comment:
Don't pass in `hass`. It will be set on the entity when it has been added to home assistant.